pkgs/historical-modules/default.nix

```
@@ -113,16 +113,14 @@ let
           (flake.deploymentModules or flake.modules).${moduleId}
         else
           flake.modules.${moduleId};
-      # Sneaky way of adding a displayVersion to a historical module:
-      # 1. build the output JSON file
-      # 2. parse the JSON file
-      # 3. add the displayVersion field
-      # 4. return a new derivation that builds the modified JSON contents
-      moduleJSON = fromJSON (builtins.unsafeDiscardStringContext (readFile module.outPath)) // {
-        inherit displayVersion;
-      };
     in
-    pkgs.writeText "replit-module-${moduleId}" (toJSON moduleJSON);
+    pkgs.stdenvNoCC.mkDerivation {
+      name = "replit-module-${moduleId}";
+      buildCommand = ''
+        set -x
+        ${pkgs.jq}/bin/jq '.displayVersion = "${displayVersion}"' < ${module} > $out
```

```suggestion
        ${pkgs.jq}/bin/jq --arg displayVersion "${displayVersion}" '.displayVersion = ''$displayVersion' < ${module} > $out
```

if we're doing this anyway. iiuc `''$` should translate to a `$` in the script that gets run. If that's the case, then the suggestion here will work.

---

I'm relying on interpolating in displayVersion from the Nix string there, not escaping it.  We could add displayVersion from Nix into the environment of the mkDerivation builder if you wanted to do it your way, but this seems simpler.

---

Yeah, I'm just trying to handle the case where `${displayVersion}` contains something unexpected that results in not-json being emitted. Unlikely, but this is why `--arg` exists, so we can be certain. It's such a small number of entries though so it should be fine.

---

Oh I see now cool!